mitodl/eslint-config
# Update @babel/eslint-parser and other plugins

### What are the relevant tickets?
Closes 
- https://github.com/mitodl/hq/issues/4847

### Description (What does it do?)
Updates eslint-config-mitodl to use `@babel/eslint-parser`  and a few other packages.

### How can this be tested?
1. Linting should pass on https://github.com/mitodl/mitxonline/pull/2282 and https://github.com/mitodl/mit-open/pull/1246


### Additional Context
A pending update is ESLint to v9 from v8. That involves a significant change to the configuration format, so i have not done it here.
